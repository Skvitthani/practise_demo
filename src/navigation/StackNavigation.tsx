import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import CameraScreen from '../screen/visionCamera/CameraScreen';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();

const StackNavigation = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{headerShown: false}}>
        <Stack.Screen name="CameraScreen" component={CameraScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default StackNavigation;
