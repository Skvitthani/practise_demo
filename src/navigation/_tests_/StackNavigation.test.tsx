import React from 'react';
import StackNavigation from '../StackNavigation';
import {render} from '@testing-library/react-native';

jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: () => ({
    Navigator: 'mocked-Navigator',
    Screen: 'mocked-Screen',
  }),
}));

jest.mock('@react-navigation/native', () => ({
  NavigationContainer: jest.fn(),
}));
jest.mock('react-native-vision-camera', () => jest.fn());
jest.mock('@react-native-camera-roll/camera-roll', () => jest.fn());
jest.mock('react-native-video', () => jest.fn());
jest.mock('react-native-linear-gradient', () => jest.fn());
jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(),
}));

describe('StackNavigation', () => {
  test('renders correctly', () => {
    const {toJSON} = render(<StackNavigation />);
    expect(toJSON()).toMatchSnapshot();
  });
});
