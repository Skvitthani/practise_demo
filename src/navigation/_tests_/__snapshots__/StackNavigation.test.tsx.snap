// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackNavigation renders correctly 1`] = `null`;
